securedrop_client/app.py

```diff
@@ -186,6 +187,14 @@ def prevent_second_instance(app: QApplication, unique_name: str) -> None:
             raise
 
 
+@contextmanager
+def thread() -> Any:
+    """Ensures that the thread is properly closed before its reference is dropped."""
+    thread = QThread()
+    yield thread
+    thread.exit()
```

Comment:
I'll make sure we wait on the thread being `finished()` before leaving the context and destroying the `QThread` instance.

Reply:
Done :white_check_mark: